File: unvisited_stones/UnvisitedStones.swift
/**
 * Находит количество валунов, которые не посетит ни одна птица.
 * 
 * @param N Количество птиц.
 * @param stones Общее количество валунов.
 * @param birds Массив с дистанциями перелёта каждой птицы.
 * @return Количество непосещённых валунов.
 */
 
func unvisitedStones(_ N: Int, _ stones: Int, _ birds: [Int]) -> Int {
    var visited = [Bool](repeating: false, count: stones + 1)
    for bird in birds {
        var pos = bird
        while pos <= stones {
            visited[pos] = true
            pos += bird
        }
    }
    return visited.dropFirst().filter { !$0 }.count
}

File: unvisited_stones/unvisited_stones.rs
/**
 * Находит количество валунов, которые не посетит ни одна птица.
 * 
 * @param n Количество птиц.
 * @param stones Общее количество валунов.
 * @param birds Массив с дистанциями перелёта каждой птицы.
 * @return Количество непосещённых валунов.
 */
 
 fn unvisited_stones(n: usize, stones: usize, birds: &[usize]) -> usize {
    let mut visited = vec![false; stones + 1];
    for &bird in birds {
        let mut pos = bird;
        while pos <= stones {
            visited[pos] = true;
            pos += bird;
        }
    }
    visited.iter().skip(1).filter(|&&v| !v).count()
}